autoload/sj/elixir.vim

```diff
@@ -39,3 +39,49 @@ function! sj#elixir#JoinDef()
   call sj#ReplaceLines(def_lineno, end_lineno, joined_line)
   return 1
 endfunction
+
+function! sj#elixir#SplitArray()
+  let [from, to] = sj#LocateBracesAroundCursor('[', ']', [
+        \ 'elixirInterpolationDelimiter',
+        \ 'elixirString',
+        \ 'elixirStringDelimiter',
+        \ 'elixirSigilDelimiter',
+        \ ])
+
+  if from < 0
+    return 0
+  endif
+
+  let items = sj#ParseJsonObjectBody(from + 1, to - 1)
```

Comment:
Trying to split an empty array, `[]`, gives me this:

```
[
]
]
```

I think the problem is that the difference between `from` and `to` needs to be at least 2 for there to be any content between them at all (I wonder how many places in the code I have where I don't check that myself :D). If the difference is smaller (as in, if it's an empty array), you could probably still go forward with the same logic, just don't call the parser and have `let items = []` or something.

Reply:
Oh my, can't believe I didn't test that case 🤦